libs/sync/operators.go

```diff
@@ -0,0 +1,171 @@
+package sync
+
+import (
+	"path"
+	"path/filepath"
+
+	"golang.org/x/exp/maps"
+)
+
+// List of operations to apply to synchronize local file systems changes to WSFS.
+type operators struct {
```

Comment:
The rename is counterintuitive to me. It's still a `diff`. Reverting also keeps the actual commit diff smaller.

Reply:
Conceptually it's not a diff though. It's a list of API calls that we would do to the WSFS. Conceptually this seemed clearer to me. But not a strong opinion and I can revert this.

Reply:
How I think about this: to go from state 1 to state 2 you apply the diff (just like for commits).

Reply:
done